Rework guide, add SSL with self-signed certs (#16153)
netdata/netdata: collectors/systemd-journal.plugin/README.md

### How do I configure a journals' centralization server?

A short summary to get journal server running can be found below.
There are two strategies you can apply, when it comes down to a centralized server for systemd journal logs.

> 1. _Active sources_, where the centralized server fetch the logs from each individual server

Reviewer: ```suggestion
1. _Active sources_, where the centralized server fetches the logs from each individual server
```

> 2. _Passive sources_, where the centralized server accept log stream from an individual server.

Reviewer: ```suggestion
2. _Passive sources_, where the centralized server accepts a log stream from an individual server.
```

For more options and reference to documentation, check `man systemd-journal-remote` and `man systemd-journal-upload`.

> We will focus on providing some instructions to setting up a _passive_ centralized server.

Reviewer: ```suggestion
We will focus on providing some instructions on setting up a _passive_ centralized server.
```

sudo systemctl daemon-reload
```

> This will make http requests as priority.

Reviewer: ```suggestion
This will make HTTP requests as priority.
```

2. Reload the daemon configs

```sh
# reload systemd
sudo systemctl daemon-reload
```

3. (OPTIONAL) If you want to change the port, edit the socket file of the `systemd-journal-remote`
```sh
# copy the service file
sudo systemctl edit systemd-journal-remote.socket
```

and add the following lines into the instructed place, and choose your desired port; save and exit.

```sh
[Socket]
ListenStream=<DESIRED_PORT>
```
    

> 4. Secure the endpoint from unauthorized access. That depends on your setup (e.g firewall setting, reverse proxies, etc)

Reviewer: Github is freaking out suggesting with code inside, here consider indenting properly the messages so they fall into the right number in the list, you can copy the raw text from this markdown, for some reason it was breaking when I was trying to suggest

2. Reload the daemon configs

   ```sh
   # reload systemd
   sudo systemctl daemon-reload
   ```

3. (OPTIONAL) If you want to change the port, edit the socket file of the `systemd-journal-remote`

   ```sh
   # copy the service file
   sudo systemctl edit systemd-journal-remote.socket
   ```

   and add the following lines into the instructed place, and choose your desired port; save and exit.

   ```sh
   [Socket]
   ListenStream=<DESIRED_PORT>
   ```

4. Secure the endpoint from unauthorized access. That depends on your setup (e.g firewall setting, reverse proxies, etc)

##### Centralization server with TLS and self-signed certificate

Follow **all** the steps from the [Centralization server without TLS (use case; only in secure intranets](#centralization-server-without-tls--use-case-only-in-secure-intranets-)

> but omit step 2. Instead of step two, make the following steps 

Reviewer: ```suggestion
but omit step 2. Instead of step two, take the following steps:
```

sudo chmod 755 /etc/ssl/ca
sudo cp "${SERVER_CN}".key /etc/ssl/private/journal-remote.key # This is not predefined but we need to clarify the key
sudo cp "${SERVER_CN}".pem /etc/ssl/certs/journal-remote.pem
sudo cp ca.pem /etc/ssl/ca/trusted.pem
```

6. Adjust the permissions for the `systemd-journal-remote` to access them.

```sh
sudo chgrp systemd-journal-remote /etc/ssl/private/journal-remote.key
sudo chgrp systemd-journal-remote /etc/ssl/certs/journal-remote.pem
sudo chgrp systemd-journal-remote /etc/ssl/ca/trusted.pem

sudo chmod 0640 /etc/ssl/private/journal-remote.key
sudo chmod 755 /etc/ssl/certs/journal-remote.pem
sudo chmod 755 /etc/ssl/ca/trusted.pem
```

6. Edit the `systemd-journal-remote.conf` to change the predefined key place and enable SSL.

sh```
sudo nano /etc/systemd/journal-remote.conf
```

You need to transform the corresponding section to something like this
```
[Remote]
Seal=false
SplitMode=host
ServerKeyFile=/etc/ssl/private/journal-remote.key
ServerCertificateFile=/etc/ssl/certs/journal-remote.pem
TrustedCertificateFile=/etc/ssl/ca/trusted.pem

> ```

Reviewer: same, add 3 spaces (or even a tab would do it, exact indentation here is 3 spaces) to the text inside each list item to indent it properly, it helps with readability

#### Configuring journal clients to push their logs to the server

In this section we will configure the clients/hosts to push their journal logs into the centralization server. You will install `systemd-journal-remote`, 
configure `systemd-journal-upload` (with or without SSL), enable and start it.

> 1. Install the `systemd-journal-remote` it run:

Reviewer: ```suggestion
1. To install `systemd-journal-remote`, run:
```

```sh
# change this according to your distro
sudo apt-get install systemd-journal-remote
```

> 2. [With SSL] Copy to the client/hosts the self-signed certificates you created before (you created one per host)

Reviewer: ```suggestion
2. **With SSL**: Copy to the client/hosts the self-signed certificates you created before (you created one per host)
```

Personal preference, I don't want to mess with brackets in markdown as they are reserved for links..

>     After this step in a folder (eg `home/user/incoming`), each host must have the following

Reviewer: ```suggestion
    After this step, each host must have the following inside a directory (e.g. `home/user/incoming`),
```

    ```
    clientX.example.com.key
    clientX.example.com.pem
    ca.pem #common between the servers
    ```

> 3. [With SSL] _On each client/host;_ create a user and a group (with the same name) called `systemd-journal-upload` 

Reviewer: ```suggestion
3. **With SSL**: _On each client/host;_ create a user and a group (with the same name) called `systemd-journal-upload` 
```

> ```sh
> sudo adduser --system --home /run/systemd --no-create-home --disabled-login --group systemd-journal-upload
> ```

Reviewer: ```suggestion
	```sh
	sudo adduser --system --home /run/systemd --no-create-home --disabled-login --group systemd-journal-upload
	```
```

> 4. [With SSL] _On each client/host;_ Navigate under the folder you placed the certificates, copy them into the expected locations (by the `systemd-journal-upload` service)

Reviewer: ```suggestion
4. **With SSL**: _On each client/host;_ Navigate under the directory you placed the certificates, copy them into the expected locations (by the `systemd-journal-upload` service)
```

> ```sh
> sudo mkdir /etc/ssl/private # make sure that you havent created this folder, and use it already, you may dont want to change it's permissions.
> sudo chmod 755 /etc/ssl/private
> sudo mkdir /etc/ssl/ca/ # make sure that you havent created this folder, and use it already, you may dont want to change it's permissions.
> sudo chmod 755 /etc/ssl/ca
> 
> cd home/user/incoming #change it accordingly, this is the place where you copied your certificates.
> sudo cp clientX.example.com.key /etc/ssl/private/journal-upload.key
> sudo cp clientX.example.com.pem /etc/ssl/certs/journal-upload.pem
> sudo cp ca.pem /etc/ssl/ca/trusted.pem
> ```

Reviewer: ```suggestion
	```sh
	sudo mkdir /etc/ssl/private # make sure that you havent created this folder, and use it already, you may dont want to change it's permissions.
	sudo chmod 755 /etc/ssl/private
	sudo mkdir /etc/ssl/ca/ # make sure that you havent created this folder, and use it already, you may dont want to change it's permissions.
	sudo chmod 755 /etc/ssl/ca
	
	cd home/user/incoming #change it accordingly, this is the place where you copied your certificates.
	sudo cp clientX.example.com.key /etc/ssl/private/journal-upload.key
	sudo cp clientX.example.com.pem /etc/ssl/certs/journal-upload.pem
	sudo cp ca.pem /etc/ssl/ca/trusted.pem
	```
```

> 5. [With SSL]  _On each client/host;_ Adjust the permission so the `systemd-journal-upload` service can access them

Reviewer: ```suggestion
5. **With SSL**:  _On each client/host;_ Adjust the permission so that the `systemd-journal-upload` service can access the files
```

> ```sh
> sudo chgrp systemd-journal-upload /etc/ssl/private/journal-upload.key
> sudo chgrp systemd-journal-upload /etc/ssl/certs/journal-upload.pem
> sudo chgrp systemd-journal-upload /etc/ssl/ca/trusted.pem
> 
> sudo chmod 0640 /etc/ssl/private/journal-upload.key
> sudo chmod 755 /etc/ssl/certs/journal-upload.pem
> sudo chmod 755 /etc/ssl/ca/trusted.pem
> ```

Reviewer: ```suggestion
	```sh
	sudo chgrp systemd-journal-upload /etc/ssl/private/journal-upload.key
	sudo chgrp systemd-journal-upload /etc/ssl/certs/journal-upload.pem
	sudo chgrp systemd-journal-upload /etc/ssl/ca/trusted.pem
	
	sudo chmod 0640 /etc/ssl/private/journal-upload.key
	sudo chmod 755 /etc/ssl/certs/journal-upload.pem
	sudo chmod 755 /etc/ssl/ca/trusted.pem
	```
```

> 6. Edit `/etc/systemd/journal-upload.conf` and set the IP address and the port of the server, like this:

Reviewer: ```suggestion
6. Edit `/etc/systemd/journal-upload.conf` and set the IP address and the port of the server, like so:
```

> ```
> [Upload]
> URL=http://centralization.server.ip:19532
> ```

Reviewer: ```suggestion
	```
	[Upload]
	URL=http://centralization.server.ip:19532
	```
```

> ⚠️ OR with SSL

Reviewer: ```suggestion
	⚠️ OR with SSL
```

> ```
> [Upload]
> URL=https://CENTRALIZED_SERVER_IP/DOMAIN :19532 #replace it accordingly
> ServerKeyFile=/etc/ssl/private/journal-upload.key
> ServerCertificateFile=/etc/ssl/certs/journal-upload.pem
> TrustedCertificateFile=/etc/ssl/ca/trusted.pem
> ```

Reviewer: ```suggestion
	```
	[Upload]
	URL=https://CENTRALIZED_SERVER_IP/DOMAIN :19532 #replace it accordingly
	ServerKeyFile=/etc/ssl/private/journal-upload.key
	ServerCertificateFile=/etc/ssl/certs/journal-upload.pem
	TrustedCertificateFile=/etc/ssl/ca/trusted.pem
	```
```
